.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,17 @@
+name: Pre-commit
+
+on:
+  workflow_dispatch:
+  workflow_call:
+
+jobs:
+  pre-commit:
+
+    timeout-minutes: 10
+    runs-on: macos-latest
```

Comment:
Do we need to run on MacOS specifically? 

I assume we'll have to deprecate it at some point, so maybe it'll be better to use Linux OS flavor

Reply:
Good catch!